### PR TITLE
Add RF antenna warning to startup warnings

### PR DESCRIPTION
**Pull request by kabturek**
Hey guys, 
i've really liked the feature on #1098 and decided to do some hacking while on a train - it's only tested in simulator as i don't have my taranis with me right now.
Would love some input and ideas if there are better ways to do it.

### CODE REVIEW

File: radio/src/myeeprom.h [hunk 1]
```diff
@@ -2152,6 +2152,7 @@ PACK(typedef struct {
   ARM_FIELD(uint8_t   ignoreSensorIds:1)
   int8_t    trimInc:3;            // Trim Increments
   uint8_t   disableThrottleWarning:1;
+  uint8_t   enableRfAntennaWarning:1;
```
Comment (schwabe) [inline radio/src/myeeprom.h]: Adding an int here will break compatibility with existing eeprom contents.

**Comment by kilrah** [comment]
There is unfortunately a truckload of implications/dependencies that are ignored/broken in here. Please specify if you're interested in learning about them and pursuing this and would like detailed explanations.

**Comment by kabturek** [comment]
thx @kilrah i'm interested in fixing this if it has any chance in being merged in future. 
I'm aware it's just simple hacking but wanted to do anything that works (small successes).

I'll rebase it on next - couldn't fetch next while i was on a train.

**Comment by kabturek** [comment]
rebased this on next but have trouble building the 2.2 source on MAC OS X. will keep trying and update the wiki when i succeed.

**Comment by kabturek** [comment]
It looks like it would be better to use ModuleData Struct (as this isnt model dependent) with its external_antenna and warn if external_antenna is true. 
And then add radio settings to enable/disable warning(RadioData struct or EXTRA_GENERAL_FIELDS) or use one of the spare fields  in Module struct for warning.

**Comment by schwabe** [comment]
@kabturek that external antenna define is only valid if the module is XJT (see the union). The moduledata hasn't that much space left but the following should work:

```
   uint8_t channelsStart;
   int8_t  channelsCount; // 0=8 channels
-  uint8_t failsafeMode:4;  //only 3 bits used
+  uint8_t failsafeMode:3;
+  uint8_t warnAntennaEnabled:1;
   uint8_t subType:3;
   uint8_t invertedSerial:1
```

**Comment by bsongis** [comment]
Please reopen when fixed, thanks!

**Comment by lypse** [comment]
Any chance on fixing this? It would be very helpful. Thanks!

**Comment by kabturek** [comment]
@lypse i didn't have time to look into this again (this PR was broken/naive/proof of concept) I have it in my todo list but i cant really promise any timing - sorry.

**Pull request closed without merging**